### DISCUSSION TITLE
is DH group ECP_384 supported in strongSwan 5.6.2

### DISCUSSION THREAD

**Discussion by chbhunia** [strongswan/strongswan#1136] [General]
Error log
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[IKE]   activating IKE_MOBIKE task
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[IKE] initiating IKE_SA ipsec-ikev2-vpn-client[1] to 172.172.172.100
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[IKE] IKE_SA ipsec-ikev2-vpn-client[1] state change: CREATED => CONNECTING
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[IKE] configured DH group ECP_384 not supported
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[MGR] checkin and destroy IKE_SA ipsec-ikev2-vpn-client[1]
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[MGR] tried to checkin and delete nonexisting IKE_SA
Jul 12 09:26:43 5GCTE-9517326ce3 ipsec[431]: 07[IKE] IKE_SA ipsec-ikev2-vpn-client[1] state change: CONNECTING => DESTROYING

Config file
:~$ sudo cat /etc/ipsec.conf
#ipsec.conf – strongSwan Ipsec configuration file.
config setup
        # you can increase the log levels to 3 and 4 to get more information
        charondebug="ike 4, knl 2, cfg 2, chd 4, net 2, esp 2, dmn 2, mgr 2"
        strictcrlpolicy=no
        uniqueids=yes
        cachecrls=no
conn ipsec-ikev2-vpn-client
     type=tunnel
     aggressive=yes
     # local ip address of outer IPSEC tunnel
      left=172.172.172.3
      # subnet for local DU F1-C addresses – static defined at DU.
      leftsubnet=1.1.1.0/16
      leftid=catapult
      leftauth=eap-mschapv2
      leftsourceip=%config
      # remote ip address of outer IPSEC tunnel
      right=172.172.172.100
      rightid=172.172.172.100    # used while creating the certificates
      # subnet for remote CU F1-C addresses
      rightsubnet=1.1.1.0/16
      rightauth=pubkey
      # defines the identity the client uses to reply to an EAP Identity request.
      eap_identity=%identity
      ike=aes256gcm16-sha384-prfsha384-ecp384! #Rakuten requirements
      esp=aes256-sha256!   #Rakuten requirements
      auto=start

**Comment by tobiasbrunner** [comment]
You need a plugin that provides the algorithm, see [here](https://docs.strongswan.org/docs/5.9/config/IKEv2CipherSuites.html#_diffie_hellman_groups) for details (although in 5.6.2 it was only the _openssl_ plugin that provided it, so make sure you have that plugin).
